server/rest/source.py

```diff
@@ -0,0 +1,115 @@
+
+    # REST Endpoints
+
+    @access.public
+    @loadmodel(map={'userId': 'user'}, model='user', level=AccessType.READ)
+    def listSources(self, user, params):
+        folder = findSourceFolder(self.getCurrentUser(), user)
+        if folder is None:
+            return []
+        else:
+            limit, offset, sort = \
+                self.getPagingParameters(params,
+                                         defaultSortDir=pymongo.DESCENDING)
+            items = [self.model('item').filter(item, self.getCurrentUser()) for
+                     item in self.model('folder').childItems(folder,
+                     limit=limit, offset=offset, sort=sort)]
+            return items
+    listSources.description = (
+        Description('List minerva sources owned by a user.')
+        .param('userId', 'User is the owner of minerva sources.',
+               required=True)
+        .param('limit', "Result set size limit (default=50).", required=False,
+               dataType='int')
+        .param('offset', "Offset into result set (default=0).", required=False,
+               dataType='int')
+        .param('sort', 'Field to sort the result list by ('
+               'default=name)', required=False)
+        .param('sortdir', "1 for ascending, -1 for descending (default=-1)",
+               required=False, dataType='int'))
+
+    @access.public
+    @loadmodel(map={'userId': 'user'}, model='user', level=AccessType.READ)
+    def getSourceFolder(self, user, params):
+        folder = findSourceFolder(self.getCurrentUser(), user)
+        return folder
+    getSourceFolder.description = (
+        Description('Get the minerva source folder owned by a user.')
+        .responseClass('Folder')
+        .param('userId', 'User is the owner of minerva sources.',
+               required=True))
+
+    @access.public
+    @loadmodel(map={'userId': 'user'}, model='user', level=AccessType.WRITE)
+    def createSourceFolder(self, user, params):
+        folder = findSourceFolder(self.getCurrentUser(), user, create=True)
+        return folder
+    createSourceFolder.description = (
+        Description('Create the minerva source folder owned by a user.')
+        .responseClass('Folder')
+        .param('userId', 'User is the owner of minerva sources.',
+               required=True))
+
+    @access.user
+    def createWmsSource(self, params):
+        # assuming to create in the user space of the current user
+        user = self.getCurrentUser()
+        folder = findSourceFolder(user, user, create=True)
+        if folder is None:
+            raise RestException('User has no Minerva Source folder.')
+        name = params['name']
+        baseURL = params['baseURL']
+        desc = 'wms source for  %s' % name
+        item = self.model('item').createItem(name, user, folder, desc)
+        minerva_metadata = {
+            'source_id': item['_id'],
+            'source_type': 'wms',
+            'wms_params': {
```

Comment:
Is 'wms_params' where username and password for authentication could be passed along?